libamqpprox/amqpprox_session.cpp

```diff
@@ -403,6 +409,46 @@ void Session::establishConnection()
         return;
     }
 
+    auto self(shared_from_this());
+    auto authResponseCb = [this,
+                           self](const AuthResponseData &authResponseData) {
+        if (authResponseData.getAuthResult() ==
+            AuthResponseData::AuthResult::DENY) {
+            LOG_ERROR << "Disconnecting unauthenticated/unauthorized client, "
+                         "reason: "
+                      << authResponseData.getReason();
+            disconnectUnauthClientGracefully();
```

Comment:
We probably want to additionally set something on SessionState, and use that to produce stats on the number of denied connections. Or alternative, pass the state into the auth intercept and let it decide how to mutate the state?